docs/presentation/Hierarchies/CustomNode.md

```diff
@@ -1,36 +1,175 @@
+[[include:Hierarchies.CustomNodeSpecification.Type.Ruleset]]
+```
+
+![Example of using "type" attribute](./media/hierarchy-with-specification-type-attribute.png)
+
+### Attribute: `label`
+
+Specifies node label. This is a string value that may be [localized](../Advanced/Localization.md).
+
+```ts
+[[include:Hierarchies.CustomNodeSpecification.Label.Ruleset]]
+```
+
+![Example of using "label" attribute](./media/hierarchy-with-specification-label-attribute.png)
+
+### Attribute: `description`
+
+> **Default value:** `""`
+
+Specifies the value of [Node.description]($presentation-common) property, which is a string that [may be localized](../Advanced/Localization.md). UI component displaying the node may choose whether and how to surface this information to users.
+
+```ts
+[[include:Hierarchies.CustomNodeSpecification.Description.Ruleset]]
+```
+
+```ts
+[[include:Hierarchies.CustomNodeSpecification.Description.Result]]
+```
+
+![Example of using "description" attribute](./media/hierarchy-with-specification-description-attribute.png)
+
+### Attribute: `imageId`
+
+> **Default value:** `""`
+
+Specifies node's image ID. If set, the ID is assigned to [Node.imageId]($presentation-common) and it's up to the UI component
+to decide what to do with it.
+
+```ts
+[[include:Hierarchies.CustomNodeSpecification.ImageId.Ruleset]]
+```
+
+```ts
+[[include:Hierarchies.CustomNodeSpecification.ImageId.Result]]
+```
+
+### Attribute: `hideNodesInHierarchy`
+
+> **Default value:** `false`
+
+When `true`, nodes produced by this specification are omitted and their children appear one hierarchy level higher.
+
+```ts
+[[include:Hierarchies.Specification.HideNodesInHierarchy.Ruleset]]
+```
+
+| `hideNodesInHierarchy: false`                                                                                                                         | `hideNodesInHierarchy: true`                                                                                                                        |
+| ----------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------- |
+| ![Example of using "hide nodes in hierarchy" attribute set to "false"](./media/hierarchy-with-specification-hidenodesinhierarchy-attribute-false.png) | ![Example of using "hide nodes in hierarchy" attribute set to "true"](./media/hierarchy-with-specification-hidenodesinhierarchy-attribute-true.png) |
+
+### Attribute: `hideIfNoChildren`
+
+> **Default value:** `false`
+
+Specifies whether the node created through this specification should be hidden if it has no child nodes.
+
+```ts
+[[include:Hierarchies.Specification.HideIfNoChildren.Ruleset]]
+```
+
+![Example of using "hide if no children" attribute](./media/hierarchy-with-specification-hideifnochildren-attribute.png)
+
```

Comment:
This attribute could use a similar screenshot matrix like `hideNodesInHierarchy` attribute has.

Reply:
added

Reply:
There is a non-obvious interaction going on with node groupings, which makes it seem that all root nodes do have children. We should either explain it or use a sample ruleset that turns off grouping.

Reply:
IMO the main problem is that class and label grouping is "on" by default. Sadly, we can't change that now... 

So at this point it's a choice of adding irrelevant attributes to make result simpler VS simpler rules and more complex result. Generally in my examples I tried to avoid adding extra attributes even if that made the result more complex. I don't have a strong opinion on which way is better - I think it needs explanation anyway.

Reply:
I changed `hideIfNoChildren` and `hideExpression` examples to use `CustomNode` for root nodes that we hide. IMO that makes the examples easier to understand.

Reply:
Thanks, the new examples do seem to be more comprehensible.

The updated `hideNodesInHierarchy` descriptions are good enough for now.